pingcap/tiup
# Fix imported pump & drainer start script

Signed-off-by: lucklove <gnu.crazier@gmail.com>



### What problem does this PR solve? 
https://github.com/pingcap/tiup/issues/797

### What is changed and how it works?
- don't add `--node-id` flag for imported instance in start scripts (to avoid node-id change)
- fetch the real node id from pd(etcd) instead of hard coding one

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)
    - case 1
        - deploy a cluster with pump
        - modify the `--node-id` flag to anything others
        - modify meta.yaml and set `imported: true` for pump
        - try to scale-in that pump server
        - destroy the cluster
    - case 2
        - deploy a cluster with pump
        - modify meta.yaml and set `imported: true` for pump
        - reload cluster
        - the generated run_pump.sh should not contain `--node-id` flag
    - case 3
        - deploy a cluster with pump
        - scale-in pump

Related changes

 - Need to cherry-pick to the release branch


Release notes:

```release-note
NONE
```
